noir_stdlib/src/collections/bounded_vec.nr

```diff
@@ -413,11 +426,19 @@ impl<T, let MaxLen: u32> BoundedVec<T, MaxLen> {
     pub fn map<U, Env>(self, f: fn[Env](T) -> U) -> BoundedVec<U, MaxLen> {
         let mut ret = BoundedVec::new();
         ret.len = self.len();
-        for i in 0..MaxLen {
-            if i < self.len() {
+
+        if is_unconstrained() {
+            for i in 0..self.len() {
                 ret.storage[i] = f(self.get_unchecked(i));
             }
+        } else {
+            for i in 0..MaxLen {
+                if i < self.len() {
```

Comment:
I wonder why it's okay to use `if i < self.len()` here, but not in `any`.

Reply:
Maybe both are fine and `exceeded_len` is an optimization? I didn't check if changing this makes a difference.

Reply:
Yes, I have no doubt both are equivalent in terms of what they do, just didn't understand why the same optimisation isn't applied here, whether it should 🤷

Reply:
Yeah, `exceeded_len` is an optimization as it's easier to check for equality than do a comparison.